### PR TITLE
Fix some simple ml helib compiler warnings

### PR DESCRIPTION
**Pull request by dimhotepus**
Just fix some common compiler warnings:
* signed | unsigned arithmetic mix.
* implicit signed values truncation (`int64_t` -> `int`).
* compute timer measure interval correctly when elapsed microseconds > `INT_MAX`.
* small performance improvements.

### CODE REVIEW

**Comment by faberga** [comment]
Hi @dubek , Can you review this PR for us, please?
Thanks.

File: DEPENDENCIES/simple_ml_helib/src/HelibCkksContext.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ void HelibCkksContext::init(unsigned long m,unsigned long r,unsigned long L) {
 
 void HelibCkksContext::init(const HelibConfig& conf) {
   config=conf;
-  unsigned long p = -1;
+  unsigned long p = std::numeric_limits<unsigned long>::max();
```
Comment (dubek) [inline DEPENDENCIES/simple_ml_helib/src/HelibCkksContext.cpp:48]: This needs to be solved in another way. In HElib `p` (at least in some places) is `long`, and `-1` is a special value which chooses CKKS scheme. We need all the arguments leading there to be `long` as well.

If you're not sure how to fix (I haven't looked at this yet) you can drop this specific change from the PR and submit without it.

File: DEPENDENCIES/simple_ml_helib/src/HelibConfig.h [hunk 2]
```diff
@@ -34,7 +35,7 @@ struct HelibConfig {
   unsigned long m;
   unsigned long r;
   unsigned long L;
-  unsigned long p=-1;
+  unsigned long p=std::numeric_limits<unsigned long>::max();
```
Comment (dubek) [inline DEPENDENCIES/simple_ml_helib/src/HelibConfig.h:38]: Same as above.

**Comment by dubek** [comment]
@faberga do you have an idea how to clean-up the issue above? It looks like helib::Context's ctor takes `unsigned long p` , but -1 is a valid value used for CKKS.

File: DEPENDENCIES/simple_ml_helib/src/HelibCkksContext.cpp (hunk 1 above)
Comment (dimhotepus) [reply DEPENDENCIES/simple_ml_helib/src/HelibCkksContext.cpp]: Temporary removed affected commits, thanks for analysis!

File: DEPENDENCIES/simple_ml_helib/src/HelibConfig.h (hunk 2 above)
Comment (dimhotepus) [reply DEPENDENCIES/simple_ml_helib/src/HelibConfig.h]: Removed commit for now.

**Comment by dimhotepus** [comment]
@faberga, @dubek 

May be you also find useful https://github.com/homenc/HElib/pull/373,
especially `PAlgebraMod` equality operator fix and fix for missed exceptions on logical errors?

**Comment by faberga** [comment]
@dubek @dimhotepus We have a new release of HElib being readied to go out this week. We might have resolved that, I will check in the morning.

**Comment by faberga** [comment]
@dubek @dimhotepus 
`unsigned long` vs `long` ... Converting a signed long to an unsigned long is something known in the C standard as a "Usual arithmetic conversion" and not considered an error per se. 

`p = -1` is used to select the CKKS scheme and although the value `-1` is passed to the `unsigned long p` in the constructor of the context, it is then used as a signed long in the constructor of zMStar, where it gets converted correctly.

We reviewed our code and decided that for now any change would be too pervasive.

-Flavio

**Comment by faberga** [comment]
Hi @dubek Can you check the changes you requested please?
Thanks.

**Comment by dimhotepus** [comment]
@dubek That's Visual Studio 2019 C++ Compiler [Warning Level 4](https://docs.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=vs-2019) and very handy [PVS Studio static code analyzer](https://www.viva64.com/en/b/0614/#ID0D321CBA0A).

**Comment by dubek** [comment]
Thanks, very nice.

**Pull request merged**